.github/workflows/presubmit.yml

```diff
@@ -21,6 +21,32 @@ jobs:
         env:
           PR_TITLE: ${{ github.event.pull_request.title }}
 
+  check_clang_tidy:
```

Comment:
1. Let's move this below `check_code_format`
2. Use Docker to set up the environment? https://github.com/taichi-dev/taichi/blob/da2fa48f4ff8e9ead8ef7246a506bff597fe3a36/.github/workflows/presubmit.yml#L94-L98

Reply:
2. Yes we can. But we can't reuse the existing Dockerfiles because `ti.test` is included. Do we need a more general Dockerfile just to setup the environment?

Reply:
> 1. Yes we can. But we can't reuse the existing Dockerfiles because `ti.test` is included. Do we need a more general Dockerfile just to setup the environment?

You can try the dockerfile in current ci/ folder. No test or build are included there.

Reply:
OK, thanks! BTW what is the usage of `ci/scripts/ubuntu_build_test`? I found current CI only invokes scripts under `.github/workflows/scripts.`.

Reply:
> OK, thanks! BTW what is the usage of `ci/scripts/ubuntu_build_test`? I found current CI only invokes scripts under `.github/workflows/scripts.`.

that is used to build and test the image generated by the dockerfile in ci folder. For CPU test, we directly use the image generated by that dockerfile (v0.1.0). For GPU test, we added some packages on top of the image built from the Dockerfile  (v0.1.1).